git-ssh.md

```diff
@@ -0,0 +1,75 @@
+# Installing SSH Key on Macbook for GitHub access
+
+I'm not going to walk through the reasoning behind using SSH keys as authorization or why GitHub is making this change. Let's just dive into the process and get the SSH keys created, installed, and referenced on GitHub.
+
+## Create the SSH Key
+1. Open terminal. **I recommend using the terminal application on your MacBook rather than the integrated terminal in VS Code.**
+1. Paste the following code into the terminal, substituting in your GitHub account email address for the `[your GitHub email]`:
+  ```
+  $ ssh-keygen -t -rsa -b 4096 -C "[your GitHub email]"
+  ```
+3. This generates a new SSH key, using the provided email as a label.
+  ```
+  > Generating public/private [your GitHub email] key pair.
+  ```
+4. When your promted to "Enter a file in which to save the key," press `Enter`. This accepts the default location.
+```
+> Enter a file in which to save the key (/Users/you/.ssh/id_youremailaddress):
+```
+5. At the prompt, type a secure passphrase. (This is optional. Using a passphrase will require you to enter the passphrase each time you attempt to authenticate with GitHub.)
+
+
+## Add your SSH Key to the ssh-agent
+**DO NOT USE A THIRD-PARTY LIBRARY TO ADD THE SSH KEY!** Use the native MacBook `ssh-add` command.
+
+1. Start the ssh-agent in the background.
+```
+$ eval "$(ssh-agent -s)"
+> Agent pid 59566
+```
+
+2. On macOS Sierra 10.12.2 or later, you need to modify your `~/.ssh/config` file to automatically load keys into the ssh-agent and store passphrases in your keychain.
+
+    1. First, check if the `~/.ssh/config` file exists:
+        ```
+          $ open ~/.ssh/config
+          > The file /Users/you/.ssh/config does not exist.
+        ```
+    1. If the file doesn't exist, create the file.
+        ```
+          $ touch ~/.ssh/config
+        ```
+    1. Open your ~/.ssh/config file, then modify the file, replacing ~/.ssh/id_ed25519 if you are not using the default location and name for your id_ed25519 key.
+        ```
+        Host *
+          AddKeysToAgent yes
+          UseKeychain yes
+          IdentityFile ~/.ssh/id_ed25519
+          ```
+      ** NOTE: If you choose to omit the passphrase to your key, omit the `UseKeychain` line.
```

Comment:
We should avoid giving them options to do thing different ways where possible. Just omit the passphrase.